var/spack/repos/builtin/packages/python/package.py

```diff
@@ -832,6 +833,12 @@ def python_include_dir(self):
 
     @property
     def site_packages_dir(self):
+        # This path is not robust and likely won't work on many systems.
+        # RHEL and Fedora like to install Python libraries to lib64
+        # instead of lib, see issues #18520 and #17126.
+        # Debian likes to install Python libraries to dist-packages
+        # instead of site-packages, see issue #24076.
+        # TODO: Figure out a better way to handle this. Can we query Python?
```

Comment:
> Can we query Python?

I've tried a few ways of doing this. For other Python vars, we use sysconfig, but that doesn't seem to work here. The following does not mention site-packages:
```console
$ python
>>> import pprint
>>> import sysconfig
>>> pprint.pprint(sysconfig.get_config_vars())
```
However, this looks more promising (https://stackoverflow.com/questions/122327/how-do-i-find-the-location-of-my-python-site-packages-directory):
```console
$ python
>>> import site
>>> site.getsitepackages()
['/Users/Adam/.spack/.spack-env/._view/zfryskidd5flrzv4sjp34m6pk6nhqu2l/lib/python3.8/site-packages']
```
We could use this and take the last 3 directories, then use that everywhere. However, this doesn't seem to work for virtualenvs. I'm also not sure how robust it will be. I can imagine some packages installing to lib/site-packages and others installing to lib64/dist-packages regardless of what this variable says.